modules/ROOT/pages/client-credentials.adoc

```diff
@@ -0,0 +1,297 @@
+The parameters related to the Client Credentials grant type are placed on a child element called <oauth-client-credentials>.
+The parameters related to ObjectStore will be placed in a child element called <oauth-store-config>.
+
+=== Configuring a Custom ObjectStore
+
+The obtained access tokens are stored in an ObjectStore. By default, the SDK stores them in the app’s default store, but users can define their own custom one, as shown in this example:
+
+[source, java, linenums]
+----
+<os:object-store name="tokenStore"
+   entryTtl="1"
+   entryTtlUnit="HOURS"
+   maxEntries="100"
+   persistent="true"
+   expirationInterval="30"
+   expirationIntervalUnit="MINUTES" />
+
+<sfdc:config name="salesforce">
+    <sfdc:client-credentials-connection display="PAGE" immediate="FALSE" prompt="CONSENT">
+        <sfdc:oauth-client-credentials clientId="${sfdc.consumerkey}" clientSecret="${sfdc.consumersecret}" tokenUrl="http://..." />
+        <sfdc:oauth-store-config objectStore="tokenStore" />
+</sfdc:config>
+----
+
+== Handling Expired Client Credentials Access Token
+
+Most service providers return (or should return) access tokens with a limited lifespan. In general, access tokens usually expire about 30 to 60 minutes after being issued, although this is not a standard that is strictly enforced by providers.
+In the same way, providers often provide a refresh token with the access token. The refresh token is a second token that can be used to replace an expired access token with a fresh one, without the need to perform the OAuth dance again.
+There is not an enforced standard that the SDK can use to automatically detect a token expiration scenario and obtain a new one. Different APIs communicate that in different ways. That is why the AccessTokenExpiredException exception exists, shown below:
+
+[source, java, linenums]
+----
+public void someOperation(@Connection SalesforceRestClient client) {
+    Response response = client.performSomeOperation();
+    if (response.getStatusCode() == 401) {
+        throw new AccessTokenExpiredException();
+    }
+}
+----
+
+The example above uses an imaginary REST client to perform an operation.
+
+* It assumes that this rest client was created through a ConnectionProvider annotated with @ClientCredentials.
+* It performs the operation and gets a Response object that contains information of the HTTP call.
+* The example assumes that a status code of 401 (UNAUTHORIZED) means that the token has expired.
+* It throws the AccessTokenExpiredException.
+
+When the SDK detects this exception, it automatically performs the refresh dance and retries the operation with a new access token.
+Note that, in practice, different APIs notify you of the access token expiration in different ways. Some give you clients that throw exceptions, while others give you custom messages. You should research the remote API to determine the best way to account for this.
```

Comment:
```suggestion
As noted earlier, different APIs notify you of an access token expiration in different ways. Some give you clients that throw exceptions, while others give you custom messages. You should research the remote API to determine the best way to account for this.
```
Consider omitting the final sentence, which is kind of vague: research the API? the best way to account for?